src/table_builder.cc

```diff
@@ -12,7 +12,31 @@ void TitanTableBuilder::Add(const Slice& key, const Slice& value) {
     return;
   }
 
-  if (ikey.type != kTypeValue || value.size() < cf_options_.min_blob_size) {
+  std::shared_ptr<VersionSet> vset;
+  if (ikey.type != kTypeValue &&
+      cf_options_.blob_run_mode == TitanBlobRunMode::kFallback &&
+      (vset = vset_ref_.lock()) != nullptr) {
```

Comment:
converting weak_ptr to shared_ptr for every key doesn't look efficient to me, though I don't know how much the penalty is. Since `vset` always outlive table factory (since `TitanDBImpl::Close()` close rocksdb before destruct itself), I think we can have hold raw pointer in `TitanTableBuilder`.

Reply:
Okay